miden/tests/integration/operations/u32_ops/arithmetic_ops.rs

```diff
@@ -754,14 +747,14 @@ proptest! {
     }
 
     #[test]
-    fn u32addc_proptest(a in any::<u32>(), b in any::<u32>(), c in 0_u32..1) {
-        let asm_op = "u32addc";
+    fn u32add3_proptest(a in any::<u32>(), b in any::<u32>(), c in any::<u32>()) {
+        let asm_op = "u32add3";
 
         let (d, overflow_b) = a.overflowing_add(b);
         let (d, overflow_c) = d.overflowing_add(c);
-        let e = if overflow_b || overflow_c { 1_u64 } else { 0_u64 };
+        let e = if overflow_b && overflow_c { 2_u64 } else if overflow_b || overflow_c { 1_u64 } else { 0_u64 };
```

Comment:
I'd probably simplify this: you can accumulate the result in a `u64` and then just split it into low and high bits.

Reply:
Thanks. Please see [c01f9c9](https://github.com/maticnetwork/miden/pull/221/commits/c01f9c9f740150f19ac3b58150701c24a1941733).